### ISSUE TITLE
[explicit-module-boundary-types] rule requires types on functions that are not directly returned

### ISSUE THREAD

**Issue by bradzacher** [typescript-eslint/typescript-eslint#2134]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/<rule>": ["<setting>"]
  }
}
```

```TS
function foo(): unknown {
  return new Proxy(apiInstance, {
    get: (target, property) => {
	  // implementation
    },
  });
}
```

**Expected Result**

No errors

**Actual Result**

```
    get: (target, property) => {
          ^^^^^^  ^^^^^^^^ Arguments are reported as untyped
```
